public/js/uploader.js

```diff
@@ -27,24 +27,42 @@ function parseFile(event) {
       })
 
       var table = $("<table/>").attr("id","preview");
-      // console.log(data);
+      var tableSize = findTableSize(data);
+
       for (var i in data) {
-          //console.log(data[i]);
           var tr="<tr>";
-          for (var key in data[i]) {
+          var key = 0;
+          while (tableSize > key) {
               //console.log(key);
-              tr+="<td>"+data[i][key]+"</td>";
+              if (typeof data[i][key] != 'undefined') {
```

Comment:
Once again, not a bad idea to save data[i] to a variable within the loop. Purely a stylistic choice to make it easier to read.